src/writer/item.cpp

```diff
@@ -33,41 +33,67 @@ namespace zim
         DefaultIndexData(const std::string& htmlData, const std::string& title)
           : title(title)
         {
+#if defined(ENABLE_XAPIAN)
           try {
             htmlParser.parse_html(htmlData, "UTF-8", true);
           } catch(...) {}
+#endif
         }
 
         bool hasIndexData() const {
+#if defined(ENABLE_XAPIAN)
           return (htmlParser.dump.find("NOINDEX") == std::string::npos);
+#else
+          return false;
+#endif
         }
 
         std::string getTitle() const {
+#if defined(ENABLE_XAPIAN)
           return zim::removeAccents(title);
-        }
+#else
+          return "";
+#endif
+         }
 
         std::string getContent() const {
+#if defined(ENABLE_XAPIAN)
           return zim::removeAccents(htmlParser.dump);
+#else
+          return "";
+#endif
         }
 
         std::string getKeywords() const {
+#if defined(ENABLE_XAPIAN)
           return zim::removeAccents(htmlParser.keywords);
+#else
+          return "";
+#endif
         }
 
         uint32_t getWordCount() const {
+#if defined(ENABLE_XAPIAN)
           return countWords(htmlParser.dump);
+#else
+          return 0;
+#endif
         }
 
         std::tuple<bool, double, double> getGeoPosition() const
         {
-          if(!htmlParser.has_geoPosition) {
-            return std::make_tuple(false, 0, 0);
+#if defined(ENABLE_XAPIAN)
+          if(htmlParser.has_geoPosition) {
+            return std::make_tuple(true, htmlParser.latitude, htmlParser.longitude);
           }
-          return std::make_tuple(true, htmlParser.latitude, htmlParser.longitude);
+#endif
+          return std::make_tuple(false, 0, 0);
         }
 
       private:
+#if defined(ENABLE_XAPIAN)
         zim::MyHtmlParser htmlParser;
+#endif
```

Comment:
Though not to be addressed immediately, I want to remind that classes with conditionally compiled data members can lead to mysterious crashes when you compile your code with one set of macros defined and link it with a library built with another set of macros (which breaks the [C++ one definition rule](https://en.wikipedia.org/wiki/One_Definition_Rule)).

Reply:
Yes, I totally agree.
This is why I want to rework the "compilation without xapian" part. As I said, libzim now compile without xapian but I'm not sure that it behaves correctly.

The plan is to "simply" remove the whole search/indexing api if libzim is compiled without xapian.
I will also fix this kind of issue in the same time.